runtime/platform_isolate_manager.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_RUNTIME_PLATFORM_ISOLATE_MANAGER_H_
+#define FLUTTER_RUNTIME_PLATFORM_ISOLATE_MANAGER_H_
+
+#include <atomic>
+#include <mutex>
+#include <unordered_set>
+
+#include "third_party/dart/runtime/include/dart_api.h"
+
+namespace flutter {
+
+/// Maintains a list of registered platform isolates, so that they can be
+/// proactively shutdown as a group during shell shutdown.
+class PlatformIsolateManager {
+ public:
+  /// Returns whether the PlatformIsolateManager is shutdown. New isolates
+  /// cannot be registered after the manager is shutdown. Callable from any
+  /// thread.
+  bool IsShutdown() const { return is_shutdown_; }
```

Comment:
Add a comment that the result may be obsolete immediately after the call except when called on the platform thread.